lib/Dialect/Moore/MooreOps.cpp

```diff
@@ -265,6 +266,61 @@ VariableOp::handlePromotionComplete(const MemorySlot &slot, Value defaultValue,
   return std::nullopt;
 }
 
+SmallVector<DestructurableMemorySlot> VariableOp::getDestructurableSlots() {
+  auto refType = getType();
```

Comment:
By ref type here do you mean the type of the value you are referring or the type of the reference itself?

Reply:
Yes, the result of the variableOp is a value refering to this variable, and the refType is the type of this value.

Reply:
There is also a structCreateOp in Moore Dialect, I am not sure @fabianschuiki suggesting whether using structCreateOp replace variableOp